packages/core/src/index.ts

```diff
@@ -187,4 +192,34 @@ export class JanusNameService {
       throw new Error(result.Errors[0].toString());
     }
   }
+
+  public async RenewDomain(
+    domain: string,
+    tld: string
+  ): Promise<RequestResult> {
+    domain = domain.toLocaleLowerCase();
+    tld = tld.toLocaleLowerCase();
+
+    const request = new RenewDomainRequest(domain, tld);
+
+    const config = Bootstrapper.Resolve<NameServiceConfig>('NameServiceConfig');
+
+    const validator = new RenewDomainValidator(config);
+    const validation = await validator.ValidateRenewDomainRequest(request);
+
+    const result = new RequestResult();
+    result.Success = validation.isValid();
+    result.Errors = validation.getFailureMessages();
+
+    if (result.Success) {
+      try {
+        const dealed = await this._jnsService.RenewDomain(request);
+        return dealed;
+      } catch (error) {
+        throw new Error(error);
```

Comment:
You probably shouldn't throw an error whose only argument is another error.

The first argument of the `Error` constructor is expected to be a `string`. If you pass an error to it, it will be automatically converted to a string through `Error#toString()`. This will cause a strange behavior in the new error ouput:
```
// ----+ See the double 'Error' prefix bellow?
//     v
Error: Error: foo
    at Object.<anonymous> (<script>.js:1:7)
    at Module._compile (internal/modules/cjs/loader.js:816:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10)
    at Module.load (internal/modules/cjs/loader.js:685:32)
    at Function.Module._load (internal/modules/cjs/loader.js:620:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:877:12)
    at internal/main/run_main_module.js:21:11
```

Since we are not doing anything at all with the error, just re-throwing it, this `try..catch` block can be removed.